openedx/features/edly/utils.py

```diff
@@ -0,0 +1,255 @@
+                                                                               absolute_urls_of_new_data)
+
+    # Operations for old xblock data
+    relative_urls_of_old_data = _get_relative_urls(old_content.get('data', []))
+    relative_urls_of_old_data, absolute_urls_of_old_data = _create_absolute_urls(xblock.location.course_key,
+                                                                                 relative_urls_of_old_data)
+    old_content_with_absolute_urls = _replace_relative_urls_with_absolute_urls(old_content.get('data'),
+                                                                               relative_urls_of_old_data,
+                                                                               absolute_urls_of_old_data)
+
+    email_params = get_email_params(xblock)
+    email_params['old_content'] = old_content_with_absolute_urls
+    email_params['new_content'] = new_content_with_absolute_urls
+    students = get_course_enrollments(xblock.location.course_key)
+    send_bulk_mail_to_students.delay(students, email_params, 'handout_changes')
+
+
+def _replace_relative_urls_with_absolute_urls(content, relative_urls, absolute_urls):
+    """
+    This function will replace the all relative url from the given content to the absolute urls
+
+    :param content: Content to be changed
+    :param relative_urls: List of relative urls to change from content
+    :param absolute_urls: List of absolute urls to change with relative urls.
+    :return: Updated content contains all absolute urls.
+    """
+    relative_urls = ['"{path}"'.format(path=path) for path in relative_urls]
+    absolute_urls = ['"{path}"'.format(path=path) for path in absolute_urls]
+    for x in range(0, len(relative_urls)):
+        content = content.replace(relative_urls[x], absolute_urls[x])
+    return content
+
+
+def _get_relative_urls(content):
+    """
+    This function will extract the relative urls from content
+
+    :param content: String from which we have to extract the relative imports
+    :return: List of relative urls
+    """
+    pattern = r'href' '*=' '*("/[:.A-z0-9/+@-]*")'
+    try:
+        relative_urls = re.findall(pattern, content)
+        relative_urls = [url.replace('"', '') for url in relative_urls]
+    except TypeError:
+        # If new course created the old_content will be None or Empty
+        return []
+    return relative_urls
+
+
+def _create_absolute_urls(course_key, relative_urls):
+    """
+    This function will make the absolute urls of the given relative urls
+
+    :param course_key: Course key for the assets
+    :param relative_urls: list of relative urls
+    :return: list of absolute urls
+    """
+    absolute_urls = []
+    for relative_url in relative_urls:
+        if _is_static_path(relative_url):
+            absolute_url = _generate_absolute_url_from_static_path(course_key, relative_url)
+            absolute_url = absolute_url.replace('block@', 'block/')
```

Comment:
Can we move this line into the function used above ?

Reply:
Done